src/Avalonia.Controls/ContentControl.cs

```diff
@@ -105,10 +117,24 @@ public VerticalAlignment VerticalContentAlignment
             get => GetValue(VerticalContentAlignmentProperty);
             set => SetValue(VerticalContentAlignmentProperty, value);
         }
+        
+        public ContainerType ContainerType
+        {
+            get => GetValue(ContainerTypeProperty);
+            set => SetValue(ContainerTypeProperty, value);
+        }
+        
+        public string? ContainerName
```

Comment:
Why is container name even different from control names? Unlike CSS we can name everything already, right? We should re-use the existing name property I think.

Reply:
CSS also has a unique element identifier called id( represented by the #). That corresponds to Name here in Avalonia. If they thought classes and ids would have been sufficient to identify containers, they wouldn't have the `container-name` property for their containment feature. 
Container name provides a mix between classes and ids/names. Unlike classes, you can only have 1 container name on a container, but unlike names, you can have multiple containers in the visual tree that have the same container. Containers are just "zones" in the view where you want to activate styles based on a query. This concept is beneficial to us, as we try to improve layout on browsers and mobile devices.